pkg/cmd/registry/artifacts/crud/get/get.go

```diff
@@ -0,0 +1,164 @@
+
+	"github.com/redhat-developer/app-services-cli/internal/config"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/registry/artifacts/util"
+
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+)
+
+type Options struct {
+	artifact   string
+	group      string
+	outputFile string
+
+	registryID string
+	version    string
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Logger     func() (logging.Logger, error)
+	Connection factory.ConnectionFunc
+	localizer  localize.Localizer
+}
+
+func NewGetCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Logger:     f.Logger,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "get",
+		Short: "Get artifact by id and group",
+		Long: `Get artifact by specifying id and group.
+Command will fetch the latest artifact from the registry based on the artifactId and group.
+
+When --version is specified command will fetch the specific version of the artifact.
+Get command will fetch artifacts based on group and artifactId and version.
+For fetching artifacts using global identifiers please use "service-registry download" command
+`,
+		Example: `
+## Get latest artifact by name
+rhoas service-registry artifacts get myschema
+
+## Get latest artifact and save its content to file
+rhoas service-registry artifacts get myschema myschema.json
+
+## Get latest artifact and pipe it to other command 
+rhoas service-registry artifacts get myschema | grep -i 'user'
+
+## Get latest artifact by specifying custom group, registry and name as flag
+rhoas service-registry artifacts get --group mygroup --registryId=myregistry --artifact myartifact
+`,
+		Args: cobra.RangeArgs(0, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.artifact = args[0]
+			}
+
+			if opts.artifact == "" {
+				return fmt.Errorf("Artifact is required. Please specify artifact as positional argument or by using --artifact flag")
+			}
+
+			if len(args) > 1 {
+				opts.outputFile = args[1]
+			}
+
+			if opts.registryID != "" {
+				return runGet(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return fmt.Errorf("No service Registry selected. Use 'rhoas service-registry use' to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+			return runGet(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.artifact, "artifact", "a", "", "Id of the artifact")
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Group of the artifact")
+	cmd.Flags().StringVarP(&opts.registryID, "registryId", "", "", "Id of the registry to be used. By default uses currently selected registry")
+	cmd.Flags().StringVarP(&opts.outputFile, "outputFile", "", "", "Filename of the output file")
```

Comment:
If it was a requirement from BU then that is okay. It seems a little weird to provide so many different ways to do the same thing though. Positional especially seems out of place - I can see a flag being suitable.

Reply:
Ah.. ok. I wanted to give number of options for User trials :D

Reply:
I did not have the full context regarding user trials, in that case it is fine to leave it in - but be aware it could make things confusing during the trials.